numba/core/typeinfer.py

```diff
@@ -53,6 +53,11 @@ def __init__(self, context, var):
 
     def add_type(self, tp, loc):
         assert isinstance(tp, types.Type), type(tp)
+        # Special case for _undef_var.
+        # If the typevar is a _undef_var, use the incoming type directly.
```

Comment:
```suggestion
        # If the typevar is the _undef_var, use the incoming type directly.
```
Presumably there's only one `types._undef_var` as it's on the `types.__init__`?